snooty/language_server.py

```diff
@@ -429,7 +429,18 @@ def m_text_document__resolve(
             )
             return str(resolved_target_path)
         elif resolveType == "directive":
-            return str(self.project.config.source_path) + fileName
+            """If the filename has a .rst extension, it might be converted from
+            a YAML file. We want to get its original file path in that case."""
+
+            # Strip the first slash from fileName so the / operator doesn't mess up :|
+            stripped_file_name = fileName[1:]
```

Comment:
Better would be `slug.lstrip("/")`, as we do in `postprocess.clean_slug()`. Fixed offset-based string manipulation just gives me the heebie jeebies :D

Reply:
yahoo thank you! but why are you reviewing during time off >:(

Reply:
shame on me!! shame!!